### ISSUE TITLE
Interval selection error on graph in Ride screen

### ISSUE THREAD

**Issue by jeffbromberger** [GoldenCheetah/GoldenCheetah#1718]
Using the "Find intervals" option I found a Peak 5s power interval of 955w.  When I select that interval it automatically highlights the part of the graph that corresponds to this.  But the highlighted area is not correct.  For example, the min of the area is 454w and the max is 845w.  In the seconds that follow the selected area the values go to 924w, 929w, 912w, 881w.  I have attached a screenshot which hopefully illustrates this better.  I'm also confused as to how the interval is labeled "955w" yet the interval stat for average power reads 961w and the highest data sample is 929w.

![ss](https://cloud.githubusercontent.com/assets/6789497/11629303/09e9731c-9cbd-11e5-9502-408b7b9410d8.png)

**Comment by liversedge** [comment]
Lets see the data in the editor .. the ride plot will be adjusted to whatever smoothing you have applied.

**Comment by jeffbromberger** [comment]
Yes the raw data looks reasonable and it selects the samples I would expect.  That's why I suspected this was an issue with the graph visualization.

![ss](https://cloud.githubusercontent.com/assets/6789497/11634721/ce3fa1cc-9cd7-11e5-895c-4d402ee0f374.png)

**Comment by liversedge** [comment]
Its not an issue with the graph visualisation -- it is plotting with the smoothing you selected.
Will close as not a bug.

Thanks.

**Comment by jeffbromberger** [comment]
There is another issue in that the power trace stops short of the right edge of the window whereas the other traces go all the way.  If I scroll the right it redraws correctly and data is in fact there.

But to the earlier question, I changed the default smoothing from 7 to 0 and now the graph looks correct (once I changed the zoom level, it did not change anything just by changing the smoothing), so I agree on that point.  But I still have to question the smoothing algorithm being completely correct.  It seems to not only smooth but also push the data later in time which creates an inaccurate selection.  If the numbers were simply changed/lowered but the peak was still selected that would be fine.  But when it is highlighting a lower set of data than the data which immediately follows it I don't see how that can be a desired result when it is labeled as a peak?
